content/kubeone/master/architecture/operating-system-manager/usage/_index.en.md

```diff
@@ -0,0 +1,145 @@
+should be rotated manually, for them to consume configurations generated by OSM.
+{{% /notice %}}
+
+To perform this rotation please follow the guide at [Rolling Restart MachineDeploments][rolling-restart].
+
+[rolling-restart]: {{< ref "../../../cheat-sheets/rollout-machinedeployment" >}}
+
+## Using custom OperatingSystemProfile
+
+The annotation `k8c.io/operating-system-profile` is used at the Machine Deployment level to select the Operating System Profile that will be used to generate configurations for provisioning the machines.
+
+### When creating a new cluster
+
+For new clusters, a dedicated terraform variable `initial_machinedeployment_operating_system_profile` was introduced for this purpose. It'll propagate the specified value to the initial machine deployments in the cluster.
+
+It can be used as follows:
+
+1. Create a custom OSP and place it in a folder inside your custom addons path, if it exists, otherwise create one. Let's take this minimalistic OSP as an example:
+
+```yaml
+apiVersion: operatingsystemmanager.k8c.io/v1alpha1
+kind: OperatingSystemProfile
+metadata:
+  name: osp-install-curl
+  namespace: kube-system
+spec:
+  osName: "ubuntu"
+  osVersion: "20.04"
+  version: "v1.0.0"
+  supportedCloudProviders:
+    - name: "aws"
+  bootstrapConfig:
+    files:
+      - path: /opt/bin/bootstrap
+        permissions: 755
+        content:
+          inline:
+            encoding: b64
+            data: |
+              #!/bin/bash
+
+              apt update && apt install -y curl jq
+
+      - path: /etc/systemd/system/bootstrap.service
+        permissions: 644
+        content:
+          inline:
+            encoding: b64
+            data: |
+              [Install]
+              WantedBy=multi-user.target
+
+              [Unit]
+              Requires=network-online.target
+              After=network-online.target
+              [Service]
+              Type=oneshot
+              RemainAfterExit=true
+              ExecStart=/opt/bin/bootstrap
+
+    modules:
+      runcmd:
+        - systemctl restart bootstrap.service
+
+  provisioningConfig:
+    files:
+      - path: /opt/hello-world
+        permissions: 644
+        content:
+          inline:
+            encoding: b64
+            data: echo "hello world"
+```
+
+**NOTE:** This is just for demonstration purposes and the custom OSPs would most likely be more complicated than this.
+
+2. Create your infrastructure
+
+```bash
+terraform apply -var=initial_machinedeployment_operating_system_profile=osp-install-curl
+```
+
+3. Now create your cluster and make sure that you are using the custom addons folder.
+
+```yaml
+apiVersion: kubeone.k8c.io/v1beta2
+kind: KubeOneCluster
+versions:
+  kubernetes: "1.22.5"
+cloudProvider:
+  aws: {}
+addons:
+  enable: true
+  path: "./custom-addons"
+```
+
+**NOTE:** The path for addons can be anything. But make sure that the custom OSP exists in that directory.
```

Comment:
> Create a custom OSP and place it in a folder inside your custom addons path, if it exists, otherwise create one

> We should use notice blocks instead of **NOTE:**
Well tbh, they add a lot of noise and in this case they are not worth the visual noise.